include/cute/atom/copy_traits_xe.hpp

```diff
@@ -1,80 +1,306 @@
-    {
-        using GTensor = Tensor<GEngine, GLayout>;
-        using Traits = Copy_Traits<Copy, GTensor>;
-        Traits traits{gtensor};
-        return Copy_Atom<Traits, typename GEngine::value_type>{traits};
-    }
+template <class CopyOp, class GTensor>
+struct XE_2D_LD_Unpack
+{
+  GTensor tensor;
+
+  using Copy_Traits = Copy_Traits<CopyOp, GTensor>;
+  template <class TS, class SLayout,
+            class TD, class DLayout>
+  CUTE_HOST_DEVICE friend constexpr void
+  copy_unpack(Copy_Traits const &traits,
+              Tensor<ViewEngine<ArithmeticTupleIterator<TS>>, SLayout> const &src,
+              Tensor<TD, DLayout> &dst)
+  {
+      static_assert(is_rmem<TD>::value);
+      int H = size<0>(traits.tensor);
+      int W = size<1>(traits.tensor) * sizeof(typename Copy_Traits::CopyInternalType);
+      auto [y, x] = src.data().coord_;
+      CopyOp::copy(traits.tensor.data().get(), W, H, W, intel::coord_t{static_cast<int>(x), static_cast<int>(y)}, &*dst.data());
+  }
+};
+
+template <class GTensor>
+struct Copy_Traits<XE_2D_U16x8x16x1x1_LD_N, GTensor>
+     : XE_2D_LD_Unpack<XE_2D_U16x8x16x1x1_LD_N, GTensor>
+{
+  // Logical thread id to thread idx
+  using ThrID = Layout<_16>;
+  // Map from (src-thr,src-val) to bit
+  using SrcLayout = Layout<Shape<_16, _64>, Stride<_0, _1>>;
+  // Map from (dst-thr,dst-val) to bit
+  using DstLayout =
+    Layout<Shape<_16, Shape<_8, _16>>, Stride<_16, Stride<_256, _1>>>;
+  // Reference map from (thr,val) to bit
+  using RefLayout = DstLayout;
+  using CopyInternalType = ushort;
+
+};
+
+template <class GTensor>
+struct Copy_Traits<XE_2D_U32x8x16x1x1_LD_N, GTensor>
+     : XE_2D_LD_Unpack<XE_2D_U32x8x16x1x1_LD_N, GTensor>
+{
+  // Logical thread id to thread idx
+  using ThrID = Layout<_16>;
+  // Map from (src-thr,src-val) to bit
+  using SrcLayout = Layout<Shape<_16, _64>, Stride<_0, _1>>;
+  // Map from (dst-thr,dst-val) to bit
+  using DstLayout =
+    Layout<Shape<_16, Shape<_8, _32>>, Stride<_32, Stride<_512, _1>>>;
+  // Reference map from (thr,val) to bit
+  using RefLayout = DstLayout;
+  using CopyInternalType = uint;
+};
+
+template <class GTensor>
+struct Copy_Traits<XE_2D_U16x8x16x4x2_LD_N, GTensor>
+     : XE_2D_LD_Unpack<XE_2D_U16x8x16x4x2_LD_N, GTensor>
+{
+  // Logical thread id to thread idx
+  using ThrID = Layout<_16>;
+  // Map from (src-thr,src-val) to bit
+  using SrcLayout = Layout<Shape<_16, _64>, Stride<_0, _1>>;
+  // Map from (dst-thr,dst-val) to bit
+  using DstLayout =
+      Layout<Shape<_16, Shape<Shape<_8, _4>, Shape<_16, _2>>>,
+             Stride<_16, Stride<Stride<_512, _4096>, Stride<_1, _256>>>>;
+  // Reference map from (thr,val) to bit
+  using RefLayout = DstLayout;
+  using CopyInternalType = ushort;
+};
+
+template <class GTensor>
+struct Copy_Traits<XE_2D_U16x8x16x2x2_LD_N, GTensor>
+     : XE_2D_LD_Unpack<XE_2D_U16x8x16x2x2_LD_N, GTensor>
+{
+  // Logical thread id to thread idx
+  using ThrID = Layout<_16>;
```

Comment:
Would it make sense to move more of this into the base class? Most of them (e.g. `ThrID`) are the same for all.